src/people.cpp

```diff
@@ -95,9 +114,20 @@ void Guests::Uninitialize()
 	this->start_voxel.y = -1;
 	this->daily_frac = 0;
 	this->next_daily_index = 0;
+
+	this->complaint_counter_hunger       = 0;
+	this->complaint_counter_thirst       = 0;
+	this->complaint_counter_waste        = 0;
+	this->complaint_counter_litter       = 0;
+	this->complaint_counter_vandalism    = 0;
+	this->time_since_complaint_hunger    = COMPLAINT_TIMEOUT;
+	this->time_since_complaint_thirst    = COMPLAINT_TIMEOUT;
+	this->time_since_complaint_waste     = COMPLAINT_TIMEOUT;
+	this->time_since_complaint_litter    = COMPLAINT_TIMEOUT;
+	this->time_since_complaint_vandalism = COMPLAINT_TIMEOUT;
```

Comment:
You added complaints about 5 different topics, and it looks like you mostly copied 5 times the same code modulo the unique variable names and constants. Did you consider making a `ComplaintTracking` class for a generic complaint, and instantiating that 5 times, possibly even as an array?

Don't quite like my name, maybe you can think of a better one.